### PR TITLE
[1LP][RFR] OSP flavor and security group

### PR DESCRIPTION
**Pull request by sshveta**
Added a test to migrate a VM with different flavor and security group in a different project in OSP and check if the migrated VM got the correct flavor and security group.

{{ pytest: cfme/tests/v2v/test_v2v_migrations_ui.py --use-provider osp13-ims --use-provider vsphere65-nested --provider-limit 2 -vvvv --long-running -k test_v2v_custom_attribute }}

### CODE REVIEW

File: cfme/tests/v2v/test_v2v_migrations_ui.py [hunk 1]
```diff
@@ -566,3 +569,74 @@ def _cleanup():
...
+    infrastructure_mapping_collection = appliance.collections.v2v_infra_mappings
+    mapping_data = mapping_data_vm_obj_single_datastore.infra_mapping_data
+
+    # Changing target cluster to other than default(admin to qe-auto)
+    component = mapping_data["clusters"][0]
+    # Removed 'admin' (default project) from component
+    component.targets.pop()
+    # Add 'qe-auto' (different project) to component
+    component.targets.append(partial_match(provider.data.clusters[1]))
+    map_cluster = component.targets[0]
```
Comment (john-dupuy) [inline cfme/tests/v2v/test_v2v_migrations_ui.py]: Q: can you do?

```suggestion
component.targets = [partial_match(provider.data.clusters[1])]
map_cluster = component.targets[0]
```

File: cfme/tests/v2v/test_v2v_migrations_ui.py [hunk 2]
```diff
@@ -566,3 +569,74 @@ def _cleanup():
...
+    # create mapping with 'qe-auto' project
+    mapping = infrastructure_mapping_collection.create(**mapping_data)
+    src_vm_obj = mapping_data_vm_obj_single_datastore.vm_list[0]
+
+    # Selecting second flavor and security group selection since first is default
+    map_flavor = provider.data.flavors[1] if attribute == "flavor" else provider.data.flavors[0]
```
Comment (john-dupuy) [inline cfme/tests/v2v/test_v2v_migrations_ui.py]: Perhaps add a small comment about what flavors these indexes correspond to in yamls

File: cfme/tests/v2v/test_v2v_migrations_ui.py [hunk 3]
```diff
@@ -566,3 +569,74 @@ def _cleanup():
...
+        infra_map=mapping.name,
+        osp_security_group=map_security_group,
+        osp_flavor=map_flavor,
+        target_provider=provider,
+        vm_list=mapping_data_vm_obj_single_datastore.vm_list,
+    )
+    assert migration_plan.wait_for_state("Started")
+    assert migration_plan.wait_for_state("In_Progress")
+    assert migration_plan.wait_for_state("Completed")
+    assert migration_plan.wait_for_state("Successful")
```
Comment (john-dupuy) [inline cfme/tests/v2v/test_v2v_migrations_ui.py]: Do we need to check each step of the way? Can we just assert success?

Comment (sshveta) [reply cfme/tests/v2v/test_v2v_migrations_ui.py]: We cannot @john-dupuy . Each of these are different methods .Only In_progress wait for plan to finish , rest all just checks plan status. So we have to go sequentially . 
Initially it was a single method that did everything but lot of tests requires only few of these states so made them individual methods.

File: cfme/tests/v2v/test_v2v_migrations_ui.py (hunk 2 above)
Comment (sshveta) [reply cfme/tests/v2v/test_v2v_migrations_ui.py]: Done

**Comment by sshveta** [comment]
> @sshveta Thoughts on moving the migration_plan creation and its steps to a fixture? Nice test case! I have some small questions

@johndupuy : Migration plan takes infrastructure mapping as parameter and in this test case flavor and security group too .Hence not making it a fixture

**Comment by mshriver** [comment]
False-positive PRT, osp13-ims didn't setup and tests were skipped. restarting.

**Comment by sshveta** [comment]
> False-positive PRT, osp13-ims didn't setup and tests were skipped. restarting.

@mshriver : OSP13-ims is being redeployed currently . Tests won't pass

File: cfme/tests/v2v/test_v2v_migrations_ui.py [hunk 4]
```diff
@@ -566,3 +569,73 @@ def _cleanup():
     view = navigate_to(migration_plan, "InProgress")
     assert not view.progress_card.is_plan_started(migration_plan.name)
     assert "no conversion host" in view.progress_card.get_error_text(migration_plan.name)
+
+
+@pytest.mark.provider([OpenStackProvider], selector=ONE_PER_VERSION,
```
Comment (mshriver) [inline cfme/tests/v2v/test_v2v_migrations_ui.py]: Thanks @sshveta !!

**Pull request merged**